pkg/agent/controller/policy/cache/rule.go

```diff
@@ -289,8 +333,11 @@ func applyCountMap(count map[string]*IPBlockItem, added, deled map[string]*IPBlo
 		if _, exist := count[ip]; !exist {
 			continue
 		}
-		count[ip].StaticCount -= del.StaticCount
-		count[ip].AgentRef.Delete(del.AgentRef.List()...)
+		if del != nil {
+			count[ip].StaticCount -= del.StaticCount
+			count[ip].AgentRef.Delete(del.AgentRef.List()...)
+			// TODO 考虑是否要删除ports
```

Comment:
Since namedPort synced from pod must not be modified. Considering other scenes or future scalability, we'd better del theme here.

Reply:
Let's assume ip and namedport one-to-one, if ip is still available then don't delete it's named port.